extension-manifest-v3/src/background/utils/map.js

```diff
@@ -69,7 +69,11 @@ export default class AutoSyncingMap {
     // and log it. A potential improvement could be to treat the
     // in-memory map as the source of truth in that scenario.)
     this._pending = new Promise((resolve, reject) => {
-      chrome.storage.local.get([this.storageKey], (result) => {
+      // Migration to session storage
+      // TODO: Remove this code after a few releases
+      chrome.storage.local.remove(this.storageKey);
```

Comment:
I wonder if it could be better in this case to move the migration out of the class? Maybe to the place where the class is initialized?

https://github.com/ghostery/ghostery-extension/blob/5607209f8297e27206852d3158cce623a0c8aa47/extension-manifest-v3/src/background/stats.js#L27

We currently lack a good place where to put migration into and the code that cleans up the storage has to live somewhere. One advantage of moving it out of the class is that AutoSyncingMap is potentially reusable, but the migration is only needed for the specific case of removing the "tabStats:v1" key.

Reply:
Current solution might have only impact on the startup, as remove is called when map is initialized. We can move it to the `stats.js` root module, but it still will be called each time background process starts.

After all, any other solution will be just for different code style. For me.keeping in the map implementation is the right place.

Reply:
We can keep it here. Though it will be called at each service worker start in both cases, since it is in the code path of the constructor.